pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -274,14 +269,14 @@ impl CustomScan for PdbScan {
             let maybe_needs_const_projections = maybe_needs_const_projections(target_list.cast());
 
             // Get all columns referenced by this RTE throughout the entire query
-            let referenced_columns = collect_maybe_fast_field_referenced_columns(root, rti);
+            let referenced_columns = collect_maybe_fast_field_referenced_columns(rti, rel);
 
             // Save the count of referenced columns for decision-making
             builder
                 .custom_private()
                 .set_referenced_columns_count(referenced_columns.len());
 
-            let ff_cnt = exec_methods::fast_fields::count_fast_fields(
+            exec_methods::fast_fields::count_fast_fields(
```

Comment:
We probably don't need this function call here anymore, except that it sets `maybe_ff`.  I think that `builder.custom_private().set_maybe_ff(!ff.is_empty());` should be moved to `collect_maybe_fast_field_referenced_columns.`